src/test/directoryLinksProvider_spec.js

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+describe("DirectoryLinksProvider API", function() {
+  "use strict";
+
+  const gMockCacheTasks = {
+    _cache: {},
+    has() {
+      return false;
+    },
+    put(url, response, cacheName) {
+      gMockCacheTasks._cache[cacheName] = {url, response};
+    },
+    match(request, cacheName) {
+      if (!request) {
+        return;
+      }
+      return gMockCacheTasks._cache[cacheName].response;
+    }
+  };
+
+  var gMockFetch = function() {
+    return {text: function() { return Promise.resolve(JSON.stringify(directoryLinks)); }};
+  };
+
+  var directoryLinks = {
+    "directory": [{
+        "directoryId": 1,
+        "title": "Mozilla Community",
+        "type": "affiliate",
+        "url": "http://contribute.mozilla.org/"
+      }, {
+        "directoryId": 2,
+        "title": "Private Browsing with Tracking Protection",
+        "type": "affiliate",
+        "url": "https://www.mozilla.org/firefox/private-browsing/?utm_source=" +
+          "directory-tiles&utm_medium=tiles&utm_content=TPV1&utm_campaign=fx-fall-15"
+      }],
+    "suggested": [{
+        "directoryId": 1,
+        "imageURI": "https://dtex4kvbppovt.cloudfront.net/images/" +
+          "f6a535cf622b97db63f4f7cdc11ac7cd7e2120f3.8723.png",
+        "enhancedImageURI": "https://dtex4kvbppovt.cloudfront.net/images/" +
+          "7c1c7a47346b9496b92d2191f38820684c54e3e4.8640.png",
+        "frecent_sites": [
+          "chat.com",
+          "fring.com",
+          "hello.firefox.com",
+          "oovoo.com",
+          "viber.com"
+        ],
+        "title": "TurboTax Suggest!",
+        "type": "sponsored",
+        "url": "https://turbotax.intuit.com/lp/ty14/bn/tmp_1_hp.jsp?znM=mind3?cid=" +
+          "bn_moz_12_hp_1855513_114761141_56708720&cvosrc=display.1855513.114761141&m_field9=nt&m_field10=56708720&",
+        "explanation": "Suggested for %1$S enthusiasts who visit sites like %2$S"
+      }]
+  };
+
+  it("should populate suggested and directory links", async(function*() {
+    // Save original CacheTasks functions and update them with mock functions.
+    var tmpHas = CacheTasks.has;
+    var tmpPut = CacheTasks.put;
+    var tmpMatch = CacheTasks.match;
+    CacheTasks.has = gMockCacheTasks.has;
```

Comment:
Again, it would be awesome here to use the real CacheTasks, because it also helps us see if the real thing works well with your code.